pkg/manager/manager.go

```diff
@@ -286,17 +292,66 @@ func (m *manager) AttachRepresentor(conf *types.PluginConf) error {
 	}
 
 	if len(conf.Trunk) > 0 {
+		log.Info().Msgf("Setting multiple VLANs for rep %s: %v", conf.Representor, conf.Trunk)
 		if err = utils.BridgeTrunkVlanAdd(m.nLink, rep, conf.Trunk); err != nil {
 			return fmt.Errorf("failed to add trunk VLAN for representor %s: %v", conf.Representor, err)
 		}
 	}
 
 	if conf.Vlan > 0 {
+		log.Info().Msgf("Setting PVID VLAN for rep %s: %d", conf.Representor, conf.Vlan)
 		if err = utils.BridgePVIDVlanAdd(m.nLink, rep, conf.Vlan); err != nil {
 			return fmt.Errorf("failed to set VLAN for representor %s: %v", conf.Representor, err)
 		}
 	}
 
+	if conf.SetUplinkVlan {
+		if err = m.addUplinkVlans(conf); err != nil {
+			return fmt.Errorf("failed to add trunk VLANs to uplink %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (m *manager) addUplinkVlans(conf *types.PluginConf) error {
+	var uplink netlink.Link
+	var err error
+
+	uplink, err = m.nLink.LinkByName(conf.PFName)
+	if err != nil {
+		return fmt.Errorf("failed to lookup PF %s: %v", conf.PFName, err)
+	}
+
+	if bonduplink, bonderr := utils.GetParentBondForLink(m.nLink, uplink); bonderr == nil {
+		log.Debug().Msgf("Using bond master for adding uplink vlans: pf:%s - master:%s",
+			uplink.Attrs().Name, bonduplink.Attrs().Name)
+		uplink = bonduplink
+	}
+
+	var vlans []int
+	if len(conf.Trunk) > 0 {
+		vlans = conf.Trunk
+	}
+
+	if conf.Vlan > 0 {
+		vlans = append(vlans, conf.Vlan)
+	}
+
+	vlanUplinkLock := flock.New(vlanUplinkLockFile)
```

Comment:
should this be made a member of manager struct as `sync.Locker` interface ?

that way, in UT this can be replaced with a mock (or a sync.Mutex) avoiding the creation of lock file in the system when tests are run WDYT ?

Reply:
I was back and forth about that.  But you've convinced me to add it to the interface.   Only catch though is `Flock.Lock` doesn't match the interface for `Sync.Locker`.   The former returns an error:

https://cs.opensource.google/go/go/+/refs/tags/go1.19.3:src/sync/mutex.go;l=40
https://github.com/gofrs/flock/blob/master/flock_unix.go#L27

Im going to go ahead and create an interface around the flock locker so its easer to unit test, and so i can just-in-time create the lockfile path if needed.

Reply:
> Only catch though is Flock.Lock doesn't match the interface for Sync.Locker.

nice catch ! was looking at this earlier today, missed that in Sync.Locker definition 🤓 

an interface around it for whats being used in code would work just fine.